central/networkgraph/entity/networktree/manager_bench_test.go

```diff
@@ -31,25 +32,39 @@ func BenchmarkCreateNetworkTree(b *testing.B) {
 		entities[i] = e
 		ip = nextIP(ip)
 	}
+	entitiesByCluster := map[string][]*storage.NetworkEntityInfo{c1: entities}
 
 	// Above data will create one of the worst possible tree since each CIDR is disjoint aka all nodes are leaves,
 	// hence resulting in comparison with every node for each new entry.
 
-	mgr := newManager()
-	b.Run("createNetworkTree", func(b *testing.B) {
-		err := mgr.Initialize(map[string][]*storage.NetworkEntityInfo{"c1": entities})
-		require.NoError(b, err)
+	b.Run("initialize", func(b *testing.B) {
+		for i := 0; i < b.N; i++ {
+			mgr := initialize(b, entitiesByCluster)
+			require.NotNil(b, mgr)
+		}
 	})
 
 	b.Run("insertIntoNetworkTree", func(b *testing.B) {
-		t := mgr.CreateNetworkTree(context.Background(), "c2")
-		require.NotNil(b, t)
-		for _, entity := range entities {
-			require.NoError(b, t.Insert(entity))
+		for i := 0; i < b.N; i++ {
+			b.StopTimer()
+			mgr := initialize(b, entitiesByCluster)
+			t := mgr.CreateNetworkTree(context.Background(), "c2")
+			b.StartTimer()
+
+			for _, entity := range entities {
+				require.NoError(b, t.Insert(entity))
+			}
 		}
 	})
 }
 
+func initialize(b *testing.B, entitiesByCluster map[string][]*storage.NetworkEntityInfo) Manager {
+	mgr := newManager()
+	err := mgr.Initialize(entitiesByCluster)
+	require.NoError(b, err)
```

Comment:
I agree but I this is to trick compiler into not optimizing this part, If we did nothing with that variable.

Reply:
I think we should keep this check as this assertion was the one that was failing.
I agree that it may impact performance but on the other hand it prevents from making mistakes thus getting fake data.

Reply:
> I think we should keep this check as this assertion was the one that was failing.

That is not good. 🤔